app/listeners/object_event_listener.rb

```diff
@@ -0,0 +1,160 @@
+
+  def self.campaign_gift_purchase_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.campaign_gift_purchase_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.event_discount_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.event_discount_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.event_discount_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_level_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_level_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_level_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_purchase_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_purchase_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_purchase_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_address_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_address_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_address_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_note_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_note_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_note_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.donation_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.donation_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.donation_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.custom_field_definition_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.custom_field_definition_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.tag_definition_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.tag_definition_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  private
+
+  def self.enqueue_transmissions_to_webhooks(event)
+    object_event_hook_configs(event).each do |config|
+      WebhookTransmitJob.perform_later(config, event)
+    end
+  end
+
+  def self.object_event_hook_configs(event)
+    nonprofit(event).object_event_hook_configs.select do |config|
+      config.object_event_types.include? event["type"]
+    end
+  end
```

Comment:
It would be good to test this somehow. We want to make sure that there's no chance the wrong webhooks are being called. I'm not quite sure the best way to do that offhand.

Reply:
Actually, I know how. Move the search for webhook configs into Nonprofit itself. Given `nonprofit` is an instance of `Nonprofit` and `type` is the event type being checked for, which of these APIs feels more natural?

* `nonprofit.find_object_event_hooks_for_type(type)` or,
* `nonprofit.object_event_hooks.for_type(type)`

Reply:
I think `nonprofit.object_event_hooks.for_type(type)` sounds better to me!

Reply:
`nonprofit.object_event_hooks_configs.for_type(type)` it is!

You can add `for_type` as an extension method described at https://guides.rubyonrails.org/association_basics.html#association-extensions

Once you do that, you can unit tests that the method is getting the correct configs

Reply:
Actually, I think it would be better if it remained on Nonprofit, so it would return all the corresponding `ObjectEventHooks`. I'd stick with the first option, in that way.

Reply:
Ahh, great then!